File: src/javaD/oop/abstractAndInterface/Animal.java
package javaD.oop.abstractAndInterface;

public abstract class Animal {
    int age = 1;
    String name = "base cat";

    public abstract String makeNoise();

    public void Poop() {
        System.out.println("The Animal Poop Now....");
    }

    public void eat() {
        System.out.println("The Animal eating Now....");
    }
}

File: src/kotlinStuff/abstractAndInterface/Animal.kt
package kotlinStuff.abstractAndInterface

abstract class Animal {
    var age = 1
    var name = "base cat"

    abstract fun makeNoise()

    open fun eye(): String { return "this cat watch things very good" }

    open fun poop() { println("The Animal Poop Now....") }

    fun eat() { println("The Animal eating Now....") }
}

